File: Backend/scheduling/src/main/java/com/draveterinaria/cl/scheduling/model/Mascota.java
package com.draveterinaria.cl.scheduling.model;

import com.fasterxml.jackson.annotation.JsonBackReference;
import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonManagedReference;
import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDate;
import java.util.Date;
import java.util.List;

@Entity
@Table(name = "MASCOTA")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@JsonIgnoreProperties({"hibernateLazyInitializer", "handler"})
public class Mascota {

    @Id
    @GeneratedValue(strategy = GenerationType.SEQUENCE, generator = "mascota_seq")
    @SequenceGenerator(name = "mascota_seq", sequenceName = "ID_MASC_SEQ", allocationSize = 1)
    @Column(name = "ID_MASCOTA")
    private Long idMascota;

    @Column(name = "NOMBRE_MASCOTA", nullable = false, length = 255)
    private String nombreMascota;

    @Column(name = "FECHA_NACIMIENTO")
    private LocalDate fechaNacimiento;

    @Column(name = "SEXO", nullable = false, length = 1)
    private String sexo;

   // @Column(name = "ID_ESPECIE", nullable = false)
   // private Long idEspecie;

    // En el campo 'tutor'
    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "ID_TUTOR", nullable = false)
    @JsonBackReference
    private Tutor tutor;

    //Relación con Especie
    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name="ID_ESPECIE", nullable = false)
    private Especie especie;

}

File: app/src/main/java/com/example/draveterinaria/data/model/Mascota.kt
package com.example.draveterinaria.data.model


import com.fasterxml.jackson.annotation.JsonBackReference
import com.fasterxml.jackson.annotation.JsonIgnoreProperties
import jakarta.persistence.*
import java.time.LocalDate

@Entity
@Table(name = "MASCOTA")
@JsonIgnoreProperties("hibernateLazyInitializer", "handler")
data class Mascota(
    // 1. Clave Primaria
    @Id
    @GeneratedValue(strategy = GenerationType.SEQUENCE, generator = "mascota_seq")
    @SequenceGenerator(name = "mascota_seq", sequenceName = "ID_MASC_SEQ", allocationSize = 1)
    @Column(name = "ID_MASCOTA")
    var idMascota: Long? = null, // Usamos 'var' para que sea mutable (setter) y 'Long?' para que sea nullable (permite null antes de ser guardado/generado)

    // 2. Propiedades Simples
    @Column(name = "NOMBRE_MASCOTA", nullable = false, length = 255)
    var nombreMascota: String,

    @Column(name = "FECHA_NACIMIENTO")
    var fechaNacimiento: LocalDate? = null, // Puede ser nullable si no es obligatorio

    @Column(name = "SEXO", nullable = false, length = 1)
    var sexo: String,

    // 3. Relación con Tutor
    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "ID_TUTOR", nullable = false)
    @JsonBackReference
    var tutor: Tutor, // Asume que la clase Tutor está definida

    // 4. Relación con Especie
    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "ID_ESPECIE", nullable = false)
    var especie: Especie // Asume que la clase Especie está definida
) {

}